tests/CIMGatewayTest.php

```diff
@@ -220,12 +220,22 @@ public function testRefundFailure()
         $response = $this->gateway->refund($this->refundOptions)->send();
 
         $this->assertFalse($response->isSuccessful());
-        $this->assertNull(
-            $response->getTransactionReference()
-        );
+        $this->assertNull($response->getTransactionReference());
         $this->assertSame(
             'The referenced transaction does not meet the criteria for issuing a credit.',
             $response->getMessage()
         );
     }
+
+    public function testShouldVoidIfRefundFailedWithResponseError54()
```

Comment:
State the error condition in the test name rather than the actual error code. Makes it more readable.